# MiMiC
Minimal Microbial Consortia creation tools


# What is MiMiC?

MiMiC proposes minimal microbial consortia from the functional potential of a given metagenomic sample. This is done by producing a binary vector (presence,absence) of all Pfams within the sample and comparing this again our pre-calculated genome database. Through an iterative process, the genomes that best match the metagenomic samples functiona repertoire are selected. Once all functions have been accounted for, or the additiona of further genome accounts for no more Pfams, the process is halted and the user provided with the list and statistics for selection.

We hope that this tool will lead to the further adoption and use of minimal microbial consortia for next generation probiotics as well as basic research into microbial communities.

# Installation

Before installating Protologger, all the dependancies below must be installed;
- [PRODIGAL](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-119)


Additional R modules used by MiMiC that must be installed are;

Once the denpendancies have been installed, the user may test the installation by running the following commands;

TBA

# Running MiMiC

TBA

# Dataset

TBA

# Citation
We ask that anyone who uses MiMiC cites not only our publication but the list of publications below which provide tools and databases which are integral for MiMiCs working;

### Tools
- [PRODIGAL](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-119)


### Databases 

